client/my-sites/stats/site.jsx

```diff
@@ -135,49 +135,101 @@ Object.defineProperty( CHART_COMMENTS, 'label', {
 
 const getActiveTab = ( chartTab ) => find( CHARTS, { attr: chartTab } ) || CHARTS[ 0 ];
 
-class StatsSite extends Component {
-	static defaultProps = {
-		chartTab: 'views',
-	};
+// Return a default amount of days to subtracts from the present day depending on the period selected.
+// Used in case no starting date is present in the URL.
+const getDefaultDaysForPeriod = ( period ) => {
+	switch ( period ) {
+		case 'hour':
+			return 1;
+		case 'day':
+			return 7;
+		case 'week':
+			return 12 * 7; // ~last 3 months
+		case 'month':
+			return 6 * 30; // ~last 6 months
+		case 'year':
+			return 5 * 365; // ~last 5 years
+		default:
+			return 30;
+	}
+};
 
-	// getDerivedStateFromProps will set the state both on init and tab switch
-	state = {
-		activeTab: null,
-		activeLegend: null,
-		customChartQuantity: null,
-	};
+function StatsBody( { siteId, chartTab = 'views', date, context, isInternal, ...props } ) {
+	const dispatch = useDispatch();
 
-	static getDerivedStateFromProps( props, state ) {
-		// when switching from one tab to another or when initializing the component,
-		// reset the active legend charts to the defaults for that tab. The legends
-		// can be then toggled on and off by the user in `onLegendClick`.
-		const activeTab = getActiveTab( props.chartTab );
-		if ( activeTab !== state.activeTab ) {
-			return {
-				activeTab,
-				// TODO: remove this when we support hourly visitors.
-				activeLegend: props.period !== 'hour' ? activeTab.legendOptions || [] : [],
-			};
-		}
-		return null;
-	}
+	const [ activeTabState, setActiveTabState ] = useState( null );
+	const [ activeLegend, setActiveLegend ] = useState( null );
+	const { period } = props.period;
+
+	const queryDate = date.format( DATE_FORMAT );
+	const moduleStrings = statsStrings();
+
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isWPAdmin = config.isEnabled( 'is_odyssey' );
+
+	const slug = useSelector( getSelectedSiteSlug );
+	const upsellModalView = useSelector(
+		( state ) => config.isEnabled( 'stats/paid-wpcom-v2' ) && getUpsellModalView( state, siteId )
+	);
+
+	const {
+		supportsPlanUsage,
+		supportsUTMStats: supportsUTMStatsFeature,
+		supportsDevicesStats: supportsDevicesStatsFeature,
+		isOldJetpack,
+		supportUserFeedback,
+	} = useSelector( ( state ) => getEnvStatsFeatureSupportChecks( state, siteId ) );
+
+	const hasSiteLoadedFeatures = useSelector(
+		( state ) => isWPAdmin && hasLoadedSiteFeatures( state, siteId )
```

Comment:
```suggestion
		( state ) => isWPAdmin || hasLoadedSiteFeatures( state, siteId )
```
Was changing the operator [from](https://github.com/Automattic/wp-calypso/pull/97838/files#diff-ea5f237b7989be993ecfab2290e70ff69102850504ef05474ce36ff414a5afacL823) `||` to `&&` intentional?

Reply:
Good catch!

Reply:
Did we still keep this change?

Reply:
Followed up on https://github.com/Automattic/wp-calypso/pull/97871.